scripts/docker/cpp-build-env.dockerfile

```diff
@@ -0,0 +1,30 @@
+# build:          docker build -t test_build_20 .
+# run:            docker run -it --rm test_build_20:latest
+# versioning:     docker tag cpp-build-env ethereum/cpp-build-env:v0.8.29.2017
+# upload:         docker push tcsiwula/cpp-build-env
+
+# get latest stable debian release image -- see https://hub.docker.com/_/debian/
+from debian:latest
+
+# maintainer email
+MAINTAINER Tim Siwula <tcsiwula@gmail.com>
+
+# by defualt, docker runs all processes as root within the container
+USER root
+
+RUN apt-get update -yq && \
+    apt-get install -yq software-properties-common && \
+    apt-get install -yq --no-install-recommends apt-utils && \
+    apt-get install -yq build-essential && \
+    apt-get install -yq libleveldb-dev && \
+    apt-get install -yq libmicrohttpd-dev && \
+# version cmake 3
+    apt-get install -yq cmake && \
+# version clang 3 -- unable to locate
+    apt-get install -yq clang && \
```

Comment:
Do not install clang at the moment. We need clang 4+ at least.

Reply:
sounds good, figure out a clean way to do this soon.